Conflict: aa137f0766f6fd674c253dd88e82fee0f8f1b721
diff --git a/deps/json11/CMakeLists.txt b/deps/json11/CMakeLists.txt
index 78cb7c94a..2c4cb8027 100644
--- a/deps/json11/CMakeLists.txt
+++ b/deps/json11/CMakeLists.txt
@@ -4,6 +4,8 @@ add_library(json11 INTERFACE)
 add_library(OBS::json11 ALIAS json11)
 
 target_include_directories(json11 INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}")
-target_compile_options(json11 INTERFACE $<$<COMPILE_LANG_AND_ID:CXX,AppleClang,Clang>:-Wno-unqualified-std-cast-call>)
+if(CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 14.0.3)
+  target_compile_options(json11 INTERFACE $<$<COMPILE_LANG_AND_ID:CXX,AppleClang,Clang>:-Wno-unqualified-std-cast-call>)
+endif()
 
 target_sources(json11 INTERFACE json11.cpp json11.hpp)
