diff --git a/cmake/Modules/CompilerConfig.cmake b/cmake/Modules/CompilerConfig.cmake
index 849d19de7..13198d1e7 100644
--- a/cmake/Modules/CompilerConfig.cmake
+++ b/cmake/Modules/CompilerConfig.cmake
@@ -108,6 +108,7 @@ else()
     - https://github.com/obsproject/obs-studio/issues/8850 for 13.1.1
   ]]
   add_compile_options(
+    -coverage
     -Werror
     -Wextra
     -Wvla
@@ -137,6 +138,8 @@ else()
     add_link_options("LINKER:$<$<PLATFORM_ID:Darwin>:-adhoc_codesign>")
   endif()
 
+  add_link_options("-coverage")
+
   if(MINGW)
     set(CMAKE_WIDL
         "widl"
diff --git a/cmake/macos/xcode.cmake b/cmake/macos/xcode.cmake
index dd304a2b4..f662691e1 100644
--- a/cmake/macos/xcode.cmake
+++ b/cmake/macos/xcode.cmake
@@ -160,3 +160,9 @@ set(CMAKE_COLOR_DIAGNOSTICS TRUE)
 set(CMAKE_SKIP_RPATH TRUE)
 # Have Xcode set default RPATH entries
 set(CMAKE_XCODE_ATTRIBUTE_LD_RUNPATH_SEARCH_PATHS "@executable_path/../Frameworks")
+
+add_compile_options(
+  $<$<NOT:$<COMPILE_LANGUAGE:Swift>>:-fprofile-arcs>
+  $<$<NOT:$<COMPILE_LANGUAGE:Swift>>:-ftest-coverage>
+)
+add_link_options(--coverage)
