diff --git a/libobs/obs-display.c b/libobs/obs-display.c
index 291000fd8..4fb24638d 100644
--- a/libobs/obs-display.c
+++ b/libobs/obs-display.c
@@ -69,6 +69,7 @@ obs_display_t *obs_display_create(const struct gs_init_data *graphics_data,
 				  uint32_t background_color)
 {
 	struct obs_display *display = bzalloc(sizeof(struct obs_display));
+	blog(LOG_INFO, "obs_display_create: Creating display=%p", display);
 
 	gs_enter_context(obs->video.graphics);
 
@@ -107,6 +108,7 @@ void obs_display_free(obs_display_t *display)
 void obs_display_destroy(obs_display_t *display)
 {
 	if (display) {
+		blog(LOG_INFO, "obs_display_destroy: Destroying display=%p", display);
 		pthread_mutex_lock(&obs->data.displays_mutex);
 		if (display->prev_next)
 			*display->prev_next = display->next;
@@ -127,6 +129,8 @@ void obs_display_resize(obs_display_t *display, uint32_t cx, uint32_t cy)
 	if (!display)
 		return;
 
+	blog(LOG_INFO, "obs_display_resize: Resizing display=%p", display);
+
 	pthread_mutex_lock(&display->draw_info_mutex);
 
 	display->next_cx = cx;
