diff --git a/.github/scripts/.build.zsh b/.github/scripts/.build.zsh
index 7670c880f..c6840e18a 100755
--- a/.github/scripts/.build.zsh
+++ b/.github/scripts/.build.zsh
@@ -237,6 +237,10 @@ ${_usage_host:-}"
           -G "${generator}"
           -DCMAKE_BUILD_TYPE:STRING=${config}
           -DCEF_ROOT_DIR:PATH="${project_root}/.deps/cef_binary_${CEF_VERSION}_${target//-/_}"
+          -DCMAKE_INSTALL_PREFIX=/usr
+          -DCMAKE_INSTALL_SYSCONFDIR=/etc
+          -DCMAKE_INSTALL_LOCALSTATEDIR=/var
+          -DCMAKE_INSTALL_LIBDIR=lib/x86_64-linux-gnu
         )
 
         local cmake_version
